### ISSUE TITLE
<M-j> mapping behaves incorrectly in completion

### ISSUE THREAD

**Issue by nhooyr** [neovim/neovim#5256]
Originally reported by a user on reddit [here](https://www.reddit.com/r/neovim/comments/4npqnx/some_help_with_mj_mapping/)

- `nvim --version`: NVIM v0.1.5-732-gc1233c4
- Vim (version: 7.4 with Included patches: 1-2152) behaves differently? Yes, just exits command line mode
- Operating system/version: MacOS Sierra
- Terminal name/version: Iterm2 3.0.7
- `$TERM`: xterm-256color

### Actual behaviour
If you map `<M-j>` to `<Down>` using `nnoremap!`, pressing it while completing exits completion instead of navigating down directories.

### Expected behaviour
It should navigate down directories

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
:noremap! <M-j> <Down>
:e <TAB><M-j>
```

Interestingly enough, if you have `wildmode=list` and `noremap! <M-k> <Up>`, and press `<M-k>` and then `<M-j>` during completion, `<M-j>` produces a `^I`(tab)?

**Comment by justinmk** [comment]
Did you try gvim or maybe macvim? In terminal Vim the behavior of `<M-...>` is completely different.

**Comment by nhooyr** [comment]
No, I only use terminal vim/neovim. Hopefully someone else can test gvim or macvim.

**Comment by justinmk** [comment]
Happens with just plain `<Down>`, no need to remap:

```
nvim -u NONE
:e <tab>
<Down>
```
